api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -383,4 +383,9 @@ message Resource {
   // Cache control properties for the resource.
   // [#not-implemented-hide:]
   CacheControl cache_control = 7;
+
+  // The Metadata field can be used to provide additional information for the resource.
+  // E.g. the trace data for debugging.
+  // [#not-implemented-hide:]
```

Comment:
Are you planning on an Envoy side implementation? If not I would probably just remove this if it's a general property of the protocol. Otherwise this is simple enough do you want to just do the implementation?

It's not exactly clear to me if this is for protocol debugging or if you expect this to get merged into the actual physical resource that it's matched with? I assume the former is if the latter it would be part of that resource type?

Reply:
There could be an Envoy side implementation, like a new extension point, but the usage could be in external package. I think it is fine to remove the hide comment.  Right, I am not expecting to merge into the actual resource right now.

Reply:
This work stream will necessitate Envoy side work; even if we're consuming in our extensions @botengyao we will need to have some record keeping as the resources are ingested. 

I agree with @mattklein123 though, since this is basically a transport protocol property and can be useful even without implementation we can go with this.